app/server/identity/identityMiddleware.ts

```diff
@@ -45,22 +45,31 @@ export const augmentRedirectURL = (
         }
       });
 
+  // To avoid potential clashes with query parameters that have a special meaning on profile (e.g. error),
+  // only forward specific query parameters.
+  const queryParametersNamesToForward = [
+    "INTCMP",
+    // Some links in payment failure emails include the user's (encrypted) email as a query parameter
+    // and/or an auto sign-in token. If present, include these in the identity redirect url,
+    // since they can be utilised to facilitate sign-in by identity frontend.
+    "encryptedEmail",
+    "autoSignInToken",
+    // By passing these profile to, can measure the sign in rates across test segments.
```

Comment:
Minor: should this be (something like):

```suggestion
    // By passing these to profile, we can measure the sign in rates across test segments.
```